src/openssl.c

```diff
@@ -4299,33 +4328,50 @@ static int pk_new(lua_State *L) {
 #endif
 		} /* switch() */
 	} else if (lua_isstring(L, 1)) {
-		int type = optencoding(L, 2, "*", X509_ANY|X509_PEM|X509_DER);
+		int format;
 		int pubonly = 0, prvtonly = 0;
-		const char *opt, *data;
+		const char *type, *data;
 		size_t len;
 		BIO *bio;
 		EVP_PKEY *pub = NULL, *prvt = NULL;
 		int goterr = 0;
 
+		if (lua_istable(L, 2)) {
+			lua_getfield(L, 2, "format");
+			lua_getfield(L, 2, "type");
+			lua_getfield(L, 2, "password");
+		} else
+			lua_pushnil(L);
```

Comment:
This branch only pushes one item vs the other branch's 3

Reply:
Now the net amount of pushed items is 1 in both branches.